src/js/utils/browser.js

```diff
@@ -5,196 +5,198 @@
 import * as Dom from './dom';
 import window from 'global/window';
 
-const USER_AGENT = window.navigator && window.navigator.userAgent || '';
-const webkitVersionMap = (/AppleWebKit\/([\d.]+)/i).exec(USER_AGENT);
-const appleWebkitVersion = webkitVersionMap ? parseFloat(webkitVersionMap.pop()) : null;
-
 /**
  * Whether or not this device is an iPod.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_IPOD = (/iPod/i).test(USER_AGENT);
+export let IS_IPOD = false;
 
 /**
  * The detected iOS version - or `null`.
  *
  * @static
- * @const
  * @type {string|null}
  */
-export const IOS_VERSION = (function() {
-  const match = USER_AGENT.match(/OS (\d+)_/i);
-
-  if (match && match[1]) {
-    return match[1];
-  }
-  return null;
-}());
+export let IOS_VERSION = null;
 
 /**
  * Whether or not this is an Android device.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_ANDROID = (/Android/i).test(USER_AGENT);
+export let IS_ANDROID = false;
 
 /**
- * The detected Android version - or `null`.
+ * The detected Android version - or `null` if not Android or indeterminable.
  *
  * @static
- * @const
  * @type {number|string|null}
  */
-export const ANDROID_VERSION = (function() {
-  // This matches Android Major.Minor.Patch versions
-  // ANDROID_VERSION is Major.Minor as a Number, if Minor isn't available, then only Major is returned
-  const match = USER_AGENT.match(/Android (\d+)(?:\.(\d+))?(?:\.(\d+))*/i);
-
-  if (!match) {
-    return null;
-  }
-
-  const major = match[1] && parseFloat(match[1]);
-  const minor = match[2] && parseFloat(match[2]);
-
-  if (major && minor) {
-    return parseFloat(match[1] + '.' + match[2]);
-  } else if (major) {
-    return major;
-  }
-  return null;
-}());
-
-/**
- * Whether or not this is a native Android browser.
- *
- * @static
- * @const
- * @type {Boolean}
- */
-export const IS_NATIVE_ANDROID = IS_ANDROID && ANDROID_VERSION < 5 && appleWebkitVersion < 537;
+export let ANDROID_VERSION;
 
 /**
  * Whether or not this is Mozilla Firefox.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_FIREFOX = (/Firefox/i).test(USER_AGENT);
+export let IS_FIREFOX = false;
 
 /**
  * Whether or not this is Microsoft Edge.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_EDGE = (/Edg/i).test(USER_AGENT);
+export let IS_EDGE = false;
 
 /**
  * Whether or not this is Google Chrome.
  *
  * This will also be `true` for Chrome on iOS, which will have different support
  * as it is actually Safari under the hood.
  *
+ * This will be true for any Chromium browser that is not Edge.
+ *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_CHROME = !IS_EDGE && ((/Chrome/i).test(USER_AGENT) || (/CriOS/i).test(USER_AGENT));
+export let IS_CHROME = false;
 
 /**
  * The detected Google Chrome version - or `null`.
+ * This has always been the _Chromium_ version, i.e. would return on Chromium Edge
```

Comment:
Maybe we should have separate `IS_CHROMIUM` and an `IS_CHROME` that is just Chrome. Is "Chromium but not Edge" useful when it captures things like Opera which may have its own quirks?

Reply:
I would expect that when people do `IS_CHROME` they are generally looking for `IS_CHROMIUM`.

Reply:
Yes, and if anything I'm surprised "Chrome but not Edge" doesn't cause problems somewhere on Edge. Added `IS_CHROMIUM` and deprecated `IS_CHROME`